ui-speedspacechart/src/components/utils.ts

```diff
@@ -49,3 +50,86 @@ export const maxPositionValues = (store: Store): MaxPositionValues => {
 export const clearCanvas = (ctx: CanvasRenderingContext2D, width: number, height: number) => {
   ctx.clearRect(0, 0, width, height);
 };
+
+/**
+ * Calculates the adaptive height based on the height supplied and the lineaires displayed.
+ * @param height - The initial height value.
+ * @param layersDisplay - The linear display options.
+ * @param isAddLayers - A flag to add or remove layers. Default is true.
+ * @returns The calculated adaptive height.
+ */
+export const getAdaptiveHeight = (
+  height: number,
+  layersDisplay: Store['layersDisplay'],
+  isAddLayers: boolean = true
```

Comment:
what about `isIncludingLinearLayers` ? (every draw is a layer, even if thez aren't in the linear)

Reply:
Good comment, thank you